docs/design-system/custom-components/application-container.md

```diff
@@ -4,44 +4,173 @@ sidebar_position: 1
+  },
+  mocks: {
+    handlers: import("../mocks/browser.js"), // Optional mock server in dev
+  },
+  storage: store, // This is your IndexedDBMethod or LocalStorageMethod instance
+});
+
+// 3. Use the "ready" event to run custom setup code
+app.on("ready", async () => {
+  const { db } = app?.storage;
+
+  // Add setup code here (like DB seeding, Dexie debug handling, etc.)
+
+  // 4. Finally, render your React app, passing "app"
+  root.render(
+    <React.StrictMode>
+      <App application={app} />
+    </React.StrictMode>,
+  );
+});
+```
+
+### The Custom Storage
+
+In this setup, `storage` is an `IndexedDBMethod` from `@nmfs-radfish/radfish`. This example is in `store.js`:
+
+```jsx
+// store.js
+
+import { IndexedDBMethod } from "@nmfs-radfish/radfish";
+
+// 1. Provide your DB name and version from environment variables
+const name = import.meta.env.VITE_INDEXED_DB_NAME;
+const version = import.meta.env.VITE_INDEXED_DB_VERSION;
+
+// 2. Define the object stores (Dexie tables)
+const stores = {
+  species: "id, name",
+  cruises: "id, cruiseName, startDate",
+  stations: "id, cruiseId, stationName",
+};
+
+// 3. Export a new IndexedDBMethod instance
+export default new IndexedDBMethod(name, version, stores);
+```
+
+Or, alternatively, set up `LocalStorageMethod`:
 
-## How It Works
+```jsx
+// store.js
 
-### Built-in Offline/Online Notifications
+import { LocalStorageMethod } from "@nmfs-radfish/radfish";
 
-The `Application` component has **built-in** functionality to check the application's network status. It displays toast notifications when the state changes:
+// 1. Provide a storage key
+const name = import.meta.env.VITE_LOCAL_STORAGE_KEY;
+
+// 2. Export a new LocalStorageMethod instance
+export default new LocalStorageMethod(name);
+```
 
-- **When the application goes offline**: A warning toast is displayed with the message `"Application is offline"`.
-- **When the application comes back online**: An info toast is displayed with the message `"Application is online"`.
+When you pass `storage: store` into the Application constructor, it knows how to read/write data from the Dexie-based IndexedDB or LocalStorage.
 
-This feature is provided **out of the box** and requires no additional setup from your side.
+For more information about configuiring offline storage, see: [State Management - Offline Storage](/building-your-application/patterns/state-management#offline-storage).
 
-### Example Usage
+### Using `<Application>` in Your React App
 
-To use the `Application` component, wrap it around your app in the entry file (e.g. `index.js` or `index.jsx`):
+When you render your React app, you can wrap everything in the `<Application>` component.
+
+By passing the `application` prop to `<Application>`, you’re effectively telling the container:
+
+> “Here’s my initialized Application instance. Use it to set up offline support, register service workers, coordinate data storage, and etc..”
 
 ```jsx
+// App.jsx
```

Comment:
It would be clearer to explain this section without using `App.js` file. It's a little too indirect. The documentation shouldn't be imposing patterns that we use in the templates.